crates/ruff_python_formatter/src/comments/placement.rs

```diff
@@ -1105,6 +1111,71 @@ fn find_only_token_in_range(
     token
 }
 
+/// Attach an enclosed end-of-line comment to a [`StmtImportFrom`].
+///
+/// For example, given:
+/// ```python
+/// from foo import (  # comment
+///    bar,
+/// )
+/// ```
+///
+/// The comment will be attached to the `StmtImportFrom` node as a dangling comment, to ensure
+/// that it remains on the same line as the `StmtImportFrom` itself.
+fn handle_enclosed_import_from_comment<'a>(
+    comment: DecoratedComment<'a>,
+    import_from: &'a StmtImportFrom,
+    locator: &Locator,
+) -> CommentPlacement<'a> {
+    let node = comment.enclosing_node();
+
+    // The comment needs to be on the same line, but before the first member, to triggering on:
+    // ```python
+    // from foo import (bar, baz,  # comment
+    //     qux,
+    // )
+    // ```
```

Comment:
This comment confuses me. It says that the comment must be on the same line, but before the first member. But the snipped shows a comment that is after the second member. Which one is it?

Reply:
The comment is accurate. In order for this path to trigger, `# comment` must be on the same line as the `import`, but before the first member. We want to trigger on:

```python
from foo import ( # bar
 baz,
)
```

But not:

```python
from foo import ( baz, # bar
)
```

Reply:
(I will clarify of course.)

Reply:
Yes, the comment shows the unformatted code for which we _don't_ want to trigger the `if` condition.

Reply:
It should be much clearer now?